src/core_modules/capture-core/components/FormFields/DateAndTime/D2Date/D2Date.component.js

```diff
@@ -1,207 +1,86 @@
-
-    containerInstance: ?HTMLElement;
-    handleTextFieldFocus: () => void;
-    handleDateSelected: (value: string) => void;
-    handleTextFieldBlur: (value: string) => void;
-    hidePopover: () => void;
-    handleDocumentClick: (event: SyntheticEvent<any>) => void;
+export class D2Date extends React.Component<Props, State> {
+    handleDateSelected: (value: {calendarDateString: string}) => void;
 
     constructor(props: Props) {
         super(props);
-
         this.state = {
-            popoverOpen: false,
+            date: props.value,
         };
-
-        this.handleTextFieldFocus = this.handleTextFieldFocus.bind(this);
         this.handleDateSelected = this.handleDateSelected.bind(this);
-        this.handleTextFieldBlur = this.handleTextFieldBlur.bind(this);
-        this.hidePopover = this.hidePopover.bind(this);
-        this.handleDocumentClick = this.handleDocumentClick.bind(this);
-    }
-
-    componentWillUnmount() {
-        // $FlowFixMe[incompatible-call] automated comment
-        document.removeEventListener('click', this.handleDocumentClick);
-    }
-
-    static propContainers = {
-        CALENDAR: 'calendar',
-        POPUP: 'popup',
-        INPUT: 'input',
-    };
-
-    handleTextFieldFocus() {
-        // $FlowFixMe[incompatible-call] automated comment
-        document.removeEventListener('click', this.handleDocumentClick);
-
-        this.setState({
-            popoverOpen: true,
-        });
-
-        this.props.onFocus && this.props.onFocus();
-    }
-
-    handleDateSelected(value: string) {
-        this.props.onBlur(value);
-        this.hidePopover();
-        this.props.onDateSelectedFromCalendar && this.props.onDateSelectedFromCalendar();
-
-        // $FlowFixMe[incompatible-call] automated comment
-        document.removeEventListener('click', this.handleDocumentClick);
-    }
-
-    handleDocumentClick(event) {
-        if ((event.target && event.target.className &&
-            event.target.className.startsWith &&
-            event.target.className.startsWith('Cal__')) ||
-            (event.target && event.target.className &&
-            event.target.className.baseVal && event.target.className.baseVal.startsWith('Cal__'))) {
-            return;
-        }
-
-        this.hidePopover();
-
-        // $FlowFixMe[incompatible-call] automated comment
-        document.removeEventListener('click', this.handleDocumentClick);
     }
 
-    handleTextFieldBlur(value: string, event) {
-        this.props.onBlur(value);
-
-        if (!event.relatedTarget || event.relatedTarget.className !== 'Cal__Container__root') {
-            this.hidePopover();
-        } else {
-            // $FlowFixMe[incompatible-call] automated comment
-            document.addEventListener('click', this.handleDocumentClick);
+    handleDateSelected(value: {calendarDateString: string}) {
+        const selectedDate = value?.calendarDateString;
+        this.setState({ date: selectedDate });
+        if (selectedDate) {
```

Comment:
There are use cases when the user wants to filter by only one date at a time. Therefore this `if` statement should only check `selectedDate !== undefined` to allow the empty string too.

<img width="384" alt="Screenshot 2024-08-12 at 10 29 51" src="https://github.com/user-attachments/assets/02885b85-e999-4185-98b8-cab5e74319ae">

Thanks!

Reply:
I don't understand the requirements here ? Empty strings wouldn't be allowed as they aren't valid `selectedDate` and the selectedDate is `undefined` when nothing passed.

Reply:
The requirement is to allow empty strings because there are valid values. In the video below, there are steps for one of the use cases when the user clears the date input and then should be able to filter the working lists.

https://github.com/user-attachments/assets/177d662d-aad8-42e0-bb02-30e9cea79737



- `if (selectedDate)` - does not allow any [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value to pass
- `if (selectedDate !== undefined)` - does not allow `undefined` to pass

Let me know if it's still unclear and we can have a call to discuss further. Thank you 😄 !

Reply:
Thanks for the video. I have update the code.